### PR TITLE
chore: POC - List v2 - Introduces currentViewItems and currentRow + Refactoring

### LINKED ISSUE #16422: List V2 POC:  Meta child widgets properties on List widget property
Make the individual properties of the widgets defined in the Entity Definition be available in the List widget.
Eg. `text`, `value` property of the Input widget be made available to the List widget.

### PR DESCRIPTION
**Pull request by ashit-rath**
## Description
Introduces 2 properties and refactors the meta widget generation logic
`currentViewItems` - Belongs to list widget which returns the current view's widget's properties
`currentRow` - special keyword to refer any other widget in the same row, Eg - `{{currentRow.Input1.text}}``

Fixes #16422

### CODE REVIEW

**Comment by Tooluloope** [comment]
@ashit-rath `currentItemsView` or `currentViewItems`

**Comment by ashit-rath** [comment]
@Tooluloope  In code i did `currentViewItems` but in description I mentioned it as `currentItemsView`. Let me update

File: app/client/src/widgets/ListWidgetV2/MetaWidgetGenerator.ts [hunk 1]
```diff
@@ -0,0 +1,675 @@
...
+};
+
+type GenerateMetaWidgetChildrenProps = {
+  index: number;
+  parentId: string;
+  templateWidget: FlattenedWidgetProps;
+  rowIndex: number;
+};
+
+type generateMetaWidgetReturn = {
```
Comment (Tooluloope) [inline app/client/src/widgets/ListWidgetV2/MetaWidgetGenerator.ts:78]: can this be `GeneratedMetaWidget`

File: app/client/src/widgets/ListWidgetV2/MetaWidgetGenerator.ts [hunk 2]
```diff
@@ -0,0 +1,675 @@
...
+
+    const key = this.getPrimaryKey(rowIndex);
+    const metaWidget = klona(templateWidget) as MetaWidget;
+    const metaCacheProps = this.getCachedMetaWidgetProps(key, templateWidgetId);
+
+    if (!metaCacheProps) {
+      return {
+        childMetaWidgets: undefined,
+        metaWidgetId: undefined,
+        metaWidgetName: undefined,
```
Comment (Tooluloope) [inline app/client/src/widgets/ListWidgetV2/MetaWidgetGenerator.ts:208]: any reason why we are returning `undefined` here and `string` on line 198?

File: app/client/src/widgets/ListWidgetV2/MetaWidgetGenerator.ts [hunk 3]
```diff
@@ -0,0 +1,675 @@
...
+  generateMetaWidgetChildren = ({
+    index,
+    parentId,
+    rowIndex,
+    templateWidget,
+  }: GenerateMetaWidgetChildrenProps) => {
+    const children: string[] = [];
+    let metaWidgets: MetaWidgets = {};
+
+    (templateWidget.children || []).map((childWidgetId: string) => {
```
Comment (Tooluloope) [inline app/client/src/widgets/ListWidgetV2/MetaWidgetGenerator.ts:263]: can we use `forEach` here instead of `map`

**Review by Tooluloope — CHANGES_REQUESTED** [review]

File: app/client/src/widgets/ListWidgetV2/widget/index.tsx [hunk 4]
```diff
@@ -167,40 +131,18 @@ class ListWidget extends BaseWidget<
       this.getMainContainer()?.bottomRow,
     );
 
-    // generate childMetaPropertyMap
-    // this.generateChildrenDefaultPropertiesMap(this.props);
-    // this.generateChildrenMetaPropertiesMap(this.props);
     this.generateChildrenEntityDefinitions(this.props);
```
[the remark this answers is not in the document]
Comment (ashit-rath) [reply app/client/src/widgets/ListWidgetV2/widget/index.tsx:134]: There is an autocomplete derived property which uses this. So I'm not 100% sure if we need it or not, we'd have to verify it's usage and then take the call.

**Review by Tooluloope — APPROVED** [review]

**Pull request merged**